src/mixins/TimeSeriesMixin.ts

```diff
@@ -68,32 +67,67 @@ export default class TimeSeriesMixin extends Mixins(PiRequestsMixin) {
       const relativeUrl = request.request.split('?')[0] + url.search
       webServiceProvider.getTimeSeriesWithRelativeUrl(relativeUrl).then( piSeries =>
       {
-        if ( piSeries.timeSeries !== undefined)
+        if ( piSeries?.timeSeries !== undefined)
         for (const timeSeries of piSeries.timeSeries) {
-          if (timeSeries.events === undefined) continue
           const resource = new SeriesUrlRequest('fews-pi', 'dummyUrl')
           const series = new Series(resource)
           const header = timeSeries.header
           if (header !== undefined) {
             const missingValue: string = header.missVal
             const timeZone = piSeries.timeZone === undefined ? 'Z' : timeZoneOffsetString(+piSeries.timeZone)
+            series.timeZone = timeZone
+            series.missingValue = missingValue
             series.header.name = `${header.stationName} - ${header.parameterId} (${header.moduleInstanceId})`
             series.header.unit = header.units
             series.header.parameter = header.parameterId
             series.header.location = header.stationName
             series.header.source = header.moduleInstanceId
             series.start = new Date(parsePiDateTime(header.startDate, timeZone) )
             series.end = new Date(parsePiDateTime(header.endDate, timeZone) )
-            series.data = timeSeries.events.map((event) => {
-              return {
-                x: new Date( parsePiDateTime(event, timeZone)),
-                y: event.value === missingValue ? null : +event.value
-              }
-            })
+            if (!elevation && timeSeries.events !== undefined) {
+              series.data = timeSeries.events.map((event) => {
+                return {
+                  x: new Date( parsePiDateTime(event, timeZone)),
+                  y: event.value === missingValue ? null : +event.value
+                }
+              })
+            } else if (elevation && timeSeries.domains !== undefined && timeSeries) {
+              series.domains = timeSeries.domains
+              this.updateTimeSeriesWithElevation(series, elevation)
+            } else {
+              throw new Error('No data found')
```

Comment:
Make the error a bit more descriptive? Maybe mention that we failed to fetch a time series because the response is empty